### PR TITLE
[Bug] Stop the intent classifier evaluation scoring four classes against the wrong logit

### LINKED ISSUE #3557: [Bug] Evaluation registry label order does not match the intent artifact
### Proposed Workgroup

Router Models & Inference Runtime

### Describe the bug

`MODEL_REGISTRY["intent"]["labels"]` in `src/training/model_eval/constants.py` lists `other`
last at index 13. The served artifact puts it at index 10, and so does
`REQUIRED_CATEGORIES` in `ft_linear_lora.py:96`. `other`, `philosophy`, `physics` and
`psychology` are all shifted.

`mom_collection_eval.py:393-400` overwrites the loaded model's `id2label` and `label2id`
with the registry order, so those four classes are scored against the wrong logit and any
intent number it reports for them is wrong.

### To Reproduce

On main at a3092868:

```bash
python -c "import sys; sys.path.insert(0,'src/training/model_eval'); \
from constants import MODEL_REGISTRY; \
print(MODEL_REGISTRY['intent']['labels'].index('other'))"
# 13
```

Against `llm-semantic-router/mmbert32k-intent-classifier-merged`, whose
`config.json` `id2label`, `category_mapping.json` and `label_mapping.json` all give:

```
9: math, 10: other, 11: philosophy, 12: physics, 13: psychology
```

### Expected behavior

Order should be preserved

### Impact

Affects intent numbers from `mom_collection_eval.py` only. The router is unaffected: the
config points `category_mapping_path` at the artifact's own `category_mapping.json`.

### Additional context

Found while reproducing #1485. A permuted order still yields plausible accuracy, so this is
invisible without an explicit comparison.

### PR DESCRIPTION
**Pull request by yaojiejia**
Closes #3557

## Purpose

`MODEL_REGISTRY["intent"]["labels"]` in `src/training/model_eval/constants.py` listed `other` last. The served artifact puts it at index 10, and so do all three trainers. `other`, `philosophy`, `physics` and `psychology` were each scored against the wrong logit.

The list is plain alphabetical everywhere else. Only the evaluation registry made an exception for `other`. This removes that exception.

Indices 0 to 9 were identical in both orders, so nothing raised. A correct prediction on the four affected categories was counted as wrong, across 3,520 of the 12,032 rows in MMLU-Pro's test split.

Also adds contract tests pinning the registry to the trainers' `REQUIRED_CATEGORIES`, which is the list that builds `label2id` and therefore decides what the head's logits mean.

## Test Plan

```bash
make test-training-contracts
make check BASE_REF=origin/main
```

## Test Result

`make test-training-contracts`: all six suites pass, including the 7 new tests.

`make check BASE_REF=origin/main`: exit 0, no failures. Runs ruff, codespell, pre-commit, `make harness-check` and `make test-training-contracts`.

Regression check: all three cases fail as intended, with 3, 5 and 9 failures respectively. Drift in either direction is caught, and so is a coordinated edit that keeps all four copies consistent.

---
<details>
<summary>Semantic Router PR Checklist</summary>

- [x] PR title begins with exactly one bracketed category, such as `[Feature]`, `[Bug]`, `[Docs]`, `[Test]`, `[Research]`, `[Community]`, or `[CI/Build]`
- [x] The title does not stack prefixes such as `[Router][Docs]`; affected modules belong in labels and the PR body
- [x] The PR links an `accepted` issue with exactly one owner: one `wg/*` label for project work or `owner/maintainers` for repository governance
- [x] Commits in this PR are signed off with `git commit -s`
- [x] The Purpose, Test Plan, and Test Result sections reflect the actual scope, commands, and blockers for this change

</details>

See [CONTRIBUTING.md](../CONTRIBUTING.md) for the full contributor workflow and commit guidance.

### CODE REVIEW

**Review by Xunzhuo — APPROVED** [review]
LGTM—this aligns the evaluator's intent labels with the served artifact and adds the right drift guards.

**Review by wilsonwu — APPROVED** [review]
LGTM
